x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/update_source_editor.js

```diff
@@ -216,6 +174,44 @@ export class UpdateSourceEditor extends Component {
           fields={this.state.tooltipFields}
         />
 
+        <EuiFormRow
+          label={i18n.translate('xpack.maps.source.esSearch.sortLabel', {
+            defaultMessage: `Sort`,
+          })}
+        >
+          <EuiFlexGroup
+            gutterSize="none"
+            justifyContent="flexEnd"
+          >
+            <EuiFlexItem>
+              <SingleFieldSelect
+                placeholder={i18n.translate(
+                  'xpack.maps.source.esSearch.sortFieldSelectPlaceholder',
+                  {
+                    defaultMessage: 'Select sort field',
+                  }
+                )}
+                value={this.props.sortField}
+                onChange={this.onSortFieldChange}
+                fields={this.state.sortFields}
+                compressed
+              />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiSelect
+                disabled={!this.props.sortField}
+                options={[
+                  { text: 'ASC', value: SORT_ORDER.ASC },
+                  { text: 'DESC', value: SORT_ORDER.DESC }
```

Comment:
I am hesitant to do this since spelling out ascending/descending will take a lot of horizontal space and shorten the sort field select. IMHO, the sort field select needs to be as long as possible to handle long field names. Spelling out asc/desc adds little value compared to more room for long field names

Reply:
Maybe I could change the select to just an up or down icon?

Reply:
A fully spelled out "Ascending" and "Descending" select box appears to be consistent with other visualizations.